WordPress/Classes/Services/MediaService.m

```diff
@@ -435,9 +435,7 @@ - (NSError *)customMediaUploadError:(NSError *)error remote:(id <MediaServiceRem
         }
     }
     if (customErrorMessage) {
-        NSMutableDictionary *userInfo = [error.userInfo mutableCopy];
-        userInfo[NSLocalizedDescriptionKey] = customErrorMessage;
-        error = [[NSError alloc] initWithDomain:error.domain code:error.code userInfo:userInfo];
+        error = [[NSError alloc] initWithDomain:error.domain code:error.code userInfo:error.key];
```

Comment:
Since the issue seems to arise from the original error's `userInfo` dictionary, this prevents it from being included when committing to Core Data. 👍 

**Some thoughts:**

- I think we can still pass in a `userInfo` dictionary that includes our custom error message.
- `error.key` will prevent compiling due to it not being a property on an [NSError](https://developer.apple.com/documentation/foundation/nserror)

Example of the userInfo that breaks:

```
{
    NSErrorFailingURLKey = "https://public-api.wordpress.com/rest/v1.1/sites/199533184/media/new?locale=en";
    NSErrorFailingURLStringKey = "https://public-api.wordpress.com/rest/v1.1/sites/199533184/media/new?locale=en";
    NSLocalizedDescription = "The Internet connection appears to be offline.";
    NSUnderlyingError = "Error Domain=kCFErrorDomainCFNetwork Code=-1009 \"(null)\" UserInfo={_NSURLErrorNWPathKey=unsatisfied (No network route), _kCFStreamErrorCodeKey=50, _kCFStreamErrorDomainKey=1}";
    "_NSURLErrorFailingURLSessionTaskErrorKey" = "LocalUploadTask <34890537-1C31-4960-AFBA-EC6924B9B8A8>.<33>";
    "_NSURLErrorRelatedURLSessionTaskErrorKey" =     (
        "LocalUploadTask <34890537-1C31-4960-AFBA-EC6924B9B8A8>.<33>"
    );
    "_kCFStreamErrorCodeKey" = 50;
    "_kCFStreamErrorDomainKey" = 1;
}
```

Perhaps we can pluck `NSErrorFailingURLKey`, `NSErrorFailingURLStringKey`, `NSLocalizedDescription` (our custom error) and use those only?

I believe `NSUnderlyingError` is the problem key / value and if we wanted to send along a description we could with `.localizedDescription`, but I'm not sure how much value that offers since at this point we know the error domain.